api/v1alpha1/contour_types.go

```diff
@@ -229,6 +229,41 @@ type EnvoyNetworkPublishing struct {
 	// +kubebuilder:validation:MaxItems=2
 	// +kubebuilder:default={{name: http, portNumber: 8080}, {name: https, portNumber: 8443}}
 	ContainerPorts []ContainerPort `json:"containerPorts,omitempty"`
+
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// +optional
+	Tolerations []Toleration `json:"tolerations,omitempty"`
```

Comment:
I'm not sure these belong under `NetworkPublishing`, we may want to move them elsewhere in the API.